src/main/java/com/ibm/cloud/sdk/core/security/IamAuthenticator.java

```diff
@@ -31,32 +31,19 @@
  * an access token from the IAM token service via the "POST /identity/token" operation.
  * When the access token expires, a new access token will be fetched.
  */
-public class IamAuthenticator extends TokenRequestBasedAuthenticator<IamToken, IamToken> implements Authenticator {
+public class IamAuthenticator extends IamRequestBasedAuthenticator implements Authenticator {
   private static final String DEFAULT_IAM_URL = "https://iam.cloud.ibm.com";
   private static final String OPERATION_PATH = "/identity/token";
-  private static final String GRANT_TYPE = "grant_type";
```

Comment:
I removed these constants because they are used only in the requestToken() function and I actually think the code is more readable and understandable WITHOUT using these contants.